core/server/src/main/java/alluxio/worker/netty/DataServerReadHandler.java

```diff
@@ -72,6 +73,9 @@
   @GuardedBy("mLock")
   private long mPosToWrite = -1;
 
+  /** Makes sure we send EOF only once. */
+  private final AtomicBoolean mEOFSent = new AtomicBoolean(false);
```

Comment:
Since the object is being reused, its less clear how this guarantees `only once`. Is there some invariant in reset?